Prototypes.lua

```diff
@@ -1786,39 +1786,63 @@ WeakAuras.event_prototypes = {
     name = L["DBM Announce"],
     init = function(trigger)
       WeakAuras.RegisterDBMCallback("DBM_Announce");
-      return "";
+      local ret = "local use_cloneId = %s;"
+      return ret:format(trigger.use_cloneId and "true" or "false");
     end,
+    statesParameter = "all",
+    canHaveAuto = true,
     args = {
       {
         name = "message",
         init = "arg",
         display = L["Message"],
-        type = "longstring"
-      }
+        type = "longstring",
+        store = true
+      },
+      {
+        name = "name",
```

Comment:
In the old code, event triggers could not return any dynamic information. With this series, all they need to do to store a bit of information, is to set "store = true" on the parameter and that information is retained in the trigger state.

For the DBM announce trigger, this means the message and icon is stored.